pyfoxsi/src/pyfoxsi/response/response.py

```diff
@@ -54,3 +78,111 @@ def number_of_telescopes(self):
     def number_of_telescopes(self, x):
         self.effective_area['total'] = self.effective_area['total'] / self.__number_of_telescopes * x
         self.__number_of_telescopes = x
+
+    @property
+    def optical_path(self):
+        return self.__optical_path
+
+    @optical_path.setter
+    def optical_path(self, x):
+        self.optical_path = x
+        self._add_optical_path_to_effective_area()
+
+    @property
+    def shutter_state(self):
+        return self.__shutter_state
+
+    @shutter_state.setter
+    def shutter_state(self, x):
+        raise AttributeError('Cannot change shutter state. Create new object with desired shutter state')
+
+    def _add_optical_path_to_effective_area(self):
+        """Add the effect of the optical path to the effective area"""
+        energies = np.array(self.optics_effective_area.index)
+        factor = np.ones(energies.shape)
+        for material in self.optical_path:
+            #effective_area = self.optics_effective_area.values
+            if material.name == pyfoxsi.detector_material:
+                factor *= factor * material.absorption(energies)
+            else:
+                factor *= factor * material.transmission(energies)
+        self.effective_area['factor'] = factor
+        self.optics_effective_area['total'] *= factor
```

Comment:
If `optics_effective_area` is intended to be just the effective area of the optics, shouldn't the factor be applied to `effective_area` instead?